sdk/monitor/azquery/custom_client.go

```diff
@@ -45,3 +48,46 @@ func NewMetricsClient(credential azcore.TokenCredential, options *MetricsClientO
 }
 
 const metricsHost string = "https://management.azure.com"
+
+// ErrorInfo - The code and message for an error.
+type ErrorInfo struct {
+	// REQUIRED; A machine readable error code.
+	Code *string `json:"code,omitempty"`
+
+	// REQUIRED; A human readable error message.
+	Message *string `json:"message,omitempty"`
+
+	// A description in detail why the operation failed.
+	RawMessage json.RawMessage
```

Comment:
Joel, is this really what we want here? I can't remember our azcore.ResponseError but I thought it had a public Code field and a String() method that returns the raw JSON; anything else is private. I think we should mimic the azcore.ResponseError type.

Reply:
resolved in person with Jeff, new format detailed below